doc/developers-guide/message_routing.md

```diff
@@ -1,59 +1,49 @@
 # Route of a message through the system
 
-In the case of a message sent from user A to user B, both of whom are served by
-the same domain, the flow of the message through the system is as described below.
+Let's examine the flow of a message sent from user A to user B, both of whom are served by the same domain.
 
-Note that hooks are called at various stages of routing - they perform many
-tasks, in fact, a lot of functionality in MongooseIM is implemented through hooks & handlers. For a general introduction
-to hooks, see [Hooks and Handlers](Hooks-and-handlers.md);
-for details of the some most important hooks, see [hooks description](hooks_description.md).
+Note that hooks are called at various stages of routing - they perform many tasks, in fact, many MongooseIM functionalities are implemented through hooks & handlers. 
+For a general introduction to hooks, see [Hooks and Handlers](Hooks-and-handlers.md); to get a closer look at a core few, see [hooks description](hooks_description.md).
 
 ## 1. Receiving the stanza
 
 User A's `ejabberd_receiver` receives the stanza and passes it to `ejabberd_c2s`.
 
 ## 2. Call to `user_send_packet`
 
-Upon some minimal validation of the stanza a hook `user_send_packet` is called.
-This is handled by a couple of modules which subscribe to this hook. Those modules
-do various complementary tasks, like storing the message in an archive, sending carbon
-copies, etc.
+Upon some minimal validation of the stanza, a hook `user_send_packet` is called.
+This is handled by a couple of modules which subscribe to this hook. 
+Those modules do various complementary tasks, like storing the message in an archive, sending carbon copies, etc.
 
 ## 3. Privacy lists and `ejabberd_router:route/3`
 
-The stanza is checked against any privacy lists in use and, in the case of being allowed,
-it will be routed by `ejabberd_router:route/3`. This also takes into account
-"blocking commands", which are part of the privacy system.
+The stanza is checked against any privacy lists in use and, in the case of being allowed, it will be routed by `ejabberd_router:route/3`. 
+This also takes into account "blocking commands", which are part of the privacy system.
 
 ## 4. Chain of routing
 
-Further on, behaviour is configurable: `ejabberd_router:route/3` passes the stanza through
-a chain of routing modules and applying `Mod:filter/3` and `Mod:route/3` from each of them.
+Further on, the behaviour is configurable: `ejabberd_router:route/3` passes the stanza through a chain of routing modules and applies `Mod:filter/3` and `Mod:route/3` from each of them.
 Each of those modules has to implement `xmpp_router` behaviour.
 
 A module can choose to either:
 
-* drop the stanza, or route the stanza
-* pass the stanza on unchanged, or modify the stanza and pass the result on
+* drop/route the stanza,
```

Comment:
Is `/` always equal to `or` in English? I'd be for using `or` here, especially in the line below.